docs/decisions/03-accounts-profiles-actors.md

```diff
@@ -0,0 +1,76 @@
+* **Profile**. A profile is the thing that is presented to the rest of the social network. This is where we manage things like handles, display names, profile pictures, follows, and followers.
+* **Actor**. An actor is how profiles are represented and transferred in the ActivityPub fediverse. This is how Letterbook will present a profile to other services, and this is how we will understand and interact with users of those other services. It presents essentially all the same information as the profile does, but in a serialized format specified by the federation protocol.
+
+I've tried to capture these relationships here:
+```mermaid
+flowchart TD
+    A1[Alice's Account]
+    A2[Bob's Account]
+    P1[Profile]
+    P2[Shared Profile]
+    P3[Profile]
+    C1["Actor \n @CmdrSpot"]
+    C2["Actor \n @Redshirts_United"]
+    C3["Actor \n @SpaceTheFinalFrontier"]
+    CR1[Remote Actor]
+    CR2[Remote Actor]
+    U1([Alice])
+    U2([Bob])
+    F(("Fediverse \n ☁️"))
+
+    U1 -->|login| A1
+    U2 -->|login| A2
+
+    subgraph "letterbook \n tenforward.example"
+        C4[/"Cached Remote Actor \n @make_it_sew@uss-enterprise.example"/]
+        C5[/"Cached Remote Actor \n @coffee_nebula@uss-voyager.example"/]
+        A1 -->|owns| P1 & P2
+        A2 -.->|access| P2
+        A2 -->|owns| P3
+        P1 --> C1
+        P2 --> C2
+        P3 --> C3    
+    end
+
+    C4 -.-> CR1
+    C5 -.-> CR2
+    C1 & C2 & C3 <-->|Activities| F
+    F <-->|Activities| CR1 & CR2
+```
+
+* <sup>1</sup> potentially more than one user, in the case of profiles that represent an organization. But that's the end, and we're at the beginning.
+* <sup>2</sup> or bot. This stuff gets complicated rapidly.
+
+## Impact
+
+The immediate result is that these models and db entities need to be created and used. On the one hand, it's probably a bit more work than simply using a singular monolithic "user" model, or similar. On the other hand, this should continue to provide good flexibility for a long time. And, it can ease development and testing, by isolating work on message handling and federation from concerns about authentication.
+
+In the longer term, this also creates the possibility that accounts can be connected to more than one profile, and that profiles can be connected to more than one account. This can enable things like easier and more robust management of bot profiles, shared access to profiles representing organizations, or delegating moderation powers between trusted contacts.
+
+## Context
+
+This permits us to separate how people log in from how they would send and receive messages. That's nice for individuals, because it gives them more flexibility to divide up their online persona, if they want to. For instance, having personal and professional profiles. The other thing this enables is to allow a single profile to be controlled by multiple accounts. And I think that is critical to organizations. And that really applies to all organizations.
```

Comment:
I'm a really big fan of this separation and I think it enables some really fun future capabilities around OAuth and SSO that I haven't seen elsewhere. Potentially we should consider how multi-account profiles will have responsibilities managed and also consider access restrictions at a per-account level: for example staff that can read and respond privately, but not make public posts, or read-only, etc. Or for example giving a Bot permission to read and post for me, but not manage the profiles settings, delete posts etc.

Also agree with where you are separating these models. Profiles and Accounts is a good slice for this.